llvm/lib/IR/Function.cpp

```diff
@@ -962,13 +962,29 @@ static std::string getMangledTypeStr(Type *Ty, bool &HasUnnamedType) {
     case Type::IntegerTyID:
       Result += "i" + utostr(cast<IntegerType>(Ty)->getBitWidth());
       break;
-    case Type::GaloisFieldTyID:
-      Result += "field";
-      break;
-    case Type::EllipticCurveTyID:
-      Result += "curve";
+    case Type::GaloisFieldTyID: {
+      switch (cast<GaloisFieldType>(Ty)->getFieldKind()) {
+        case GaloisFieldKind::GALOIS_FIELD_BLS12381_BASE:     Result += "field_bls12381_base";     break;
+        case GaloisFieldKind::GALOIS_FIELD_BLS12381_SCALAR:   Result += "field_bls12381_scalar";   break;
+        case GaloisFieldKind::GALOIS_FIELD_CURVE25519_BASE:   Result += "field_curve25519_base";   break;
+        case GaloisFieldKind::GALOIS_FIELD_CURVE25519_SCALAR: Result += "field_curve25519_scalar"; break;
+        case GaloisFieldKind::GALOIS_FIELD_PALLAS_BASE:       Result += "field_pallas_base";       break;
+        case GaloisFieldKind::GALOIS_FIELD_PALLAS_SCALAR:     Result += "field_pallas_scalar";     break;
+        case GaloisFieldKind::GALOIS_FIELD_VESTA_BASE:        Result += "field_vesta_base";        break;
+        case GaloisFieldKind::GALOIS_FIELD_VESTA_SCALAR:      Result += "field_vesta_scalar";      break;
+      }
```

Comment:
Add `default`.

Reply:
Since we cover all variants of the enum, do we really need a default here? It will result in a warning:
```
warning: default label in switch which covers all enumeration values
```